dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,312 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBox.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+
+using namespace std;
+
+NumberBox::NumberBox()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_NumberBox);
+    SetDefaultStyleKey(this);
+}
+
+
+void NumberBox::OnApplyTemplate()
+{
+    // Initializations - Visual Components
+    winrt::IControlProtected controlProtected = *this;
+    m_TextBox = GetTemplateChildT<winrt::TextBox>(L"InputBox", controlProtected);
+    m_SpinDown = GetTemplateChildT<winrt::Button>(L"DownSpinButton", controlProtected);
+    m_SpinUp = GetTemplateChildT<winrt::Button>(L"UpSpinButton", controlProtected);
+
+    // Initializations - Visual States
+    SetSpinButtonVisualState();
+    SetHeader();
+
+    // Initializations - Interactions
+    m_SpinDown.Click({ this, &NumberBox::OnSpinDownClick });
+    m_SpinUp.Click({ this, &NumberBox::OnSpinUpClick });
+    m_TextBox.KeyUp({ this, &NumberBox::OnNumberBoxKeyUp });
+    this->PointerWheelChanged({ this, &NumberBox::OnScroll });
+
+    // Initializations - Tools, etc.
+    Formatter = winrt::DecimalFormatter();
+    IRounder = winrt::IncrementNumberRounder();
+    SRounder = winrt::SignificantDigitsNumberRounder();
+    UpdateFormatter();
+    UpdateRounder();
+    // Set Text to reflect preset Value
+    if (s_ValueProperty != DEFAULTVALUE && m_TextBox) {
+        UpdateTextToValue();
+    }
+
+    // Register LostFocus Event
+    if ( m_TextBox ) {
+        m_TextBox.LostFocus({ this, &NumberBox::OnTextBoxLostFocus });
+    }
+}
+
+void  NumberBox::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
```

Comment:
switch to per-property property changed callbacks (see TabView for the new pattern) -- you switch by removing this line from the IDL:

[MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnPropertyChanged")]